sdk/go/api_esc_extensions.go

```diff
@@ -29,13 +33,97 @@ func NewAuthContext(accessToken string) context.Context {
 	)
 }
 
+// DefaultAuthContext creates a new context, retrieving Pulumi Access Token
+// from either PULUMI_ACCESS_TOKEN environment variable, or from
+// currently logged in account in Pulumi CLI or ESC CLI
+//
+// This context can be used to authenticate requests to the ESC API.
+func NewDefaultAuthContext() (context.Context, error) {
+	accessToken := os.Getenv("PULUMI_ACCESS_TOKEN")
+	if accessToken != "" {
+		return NewAuthContext(accessToken), nil
+	}
+
+	workspace := esc_workspace.New(esc_workspace.DefaultFS(), esc_workspace.DefaultPulumiWorkspace())
+	account, _, err := workspace.GetCurrentAccount(false)
```

Comment:
Should `GetCurrentAccount(false)` be `GetCurrentAccount(true)` to ensure it's falling back to the Pulumi CLI credentials when the ESC credentials have not been set?